accounts-db/src/buffered_reader.rs

```diff
@@ -0,0 +1,230 @@
+//! File I/O buffered reader for AppendVec
+use {
+    crate::{append_vec::ValidSlice, file_io::read_more_buffer},
+    std::{fs::File, ops::Range},
+};
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum BufferedReaderStatus {
+    Eof,
+    Success,
+}
+
+/// read a file a large buffer at a time and provide access to a slice in that buffer
```

Comment:
I've been editing this comment for nearly half an hour! 

At first I was going to suggest that this should be a `std::io::BufRead` implementation that uses `read_at` to read, and then we could use `std::io::BufReader` with it.

Then I figured that the issue is that when account data is read, it must be read fully and be contiguous in memory since that's how callers expect it, and that's not something BufReader/BufRead guarantee.

I think the docs should say something like:

```
Specialized BufRead-like type for reading account data.

Callers can use this type to iterate efficiently over append vecs. They can do so by repeatedly
calling read(), advance_offset() and set_required_data_len(account_data_len) once the next account
data length is known.

Unlike BufRead/BufReader, this type guarnatees that on the next read() after calling
set_required_data_len(len), the whole account data is buffered _linearly_ in memory and available to
be returned.
```

Reply:
I added these at the top.